src/dns.h

```diff
@@ -588,26 +605,74 @@ hsk_dns_txt_alloc(void);
 void
 hsk_dns_txt_free(hsk_dns_txt_t *txt);
 
+/**
+ * Returns: length of uncompressed name
+ * In:      data_:      pointer to the beginning of a compressed wire-formatted
+ *                      name in a DNS message
+ *          data_len_:  pointer to remaining length of data in the `data_`
+ *                      byte array
+ *          dmp:        complete copy of the DNS message for pointer reference
+ * Out:     name:       (optional) the uncompressed name read from `data_`
+ *                      as a byte array, serialized in DNS wire format
+ */
 int
 hsk_dns_name_parse(
   uint8_t **data_,
   size_t *data_len_,
   const hsk_dns_dmp_t *dmp,
-  char *name
+  uint8_t *name
+);
+
+/**
+ * Returns: boolean (success)
+ * In:      name:  pointer to uncompressed wire-formatted name in a byte array
+ * Out:     data:  (optional) pointer to beginning of destination for compressed
+ *                 name in a DNS message serialized as a byte array
+ *          len:   length of data written (compressed length)
+ *          cmp:   (optional) map of labels and their position in the DNS
+ *                 message for label compression
+ */
+static bool
```

Comment:
If the function is static, it's private to that file. So it should not be part of header.